tests/engine/test_filter_evttype_resolver.cpp

```diff
@@ -214,7 +224,7 @@ TEST_CASE("Should find event types from filter", "[rule_loader]")
     SECTION("non_evt_type_or_nested_multi_not")
     {
         auto f = compile("evt.type=openat or not (evt.type=close and proc.name=nginx)");
-        compare_evttypes(std::move(f), not_close);
+        compare_evttypes(std::move(f), all_events);
```

Comment:
This test is where we were conceptually all wrong. In this case, the negated condition actually matches every event types, including `close`.